pkg/kubelet/rkt/pod_info.go

```diff
@@ -0,0 +1,138 @@
+	// TODO(yifan): Should be appName to exit code in the future.
+	exitCodes map[string]int
+}
+
+func newPodInfo() *podInfo {
+	return &podInfo{pid: -1, exitCodes: make(map[string]int)}
+}
+
+// parseStatus parses the result of 'rkt status' into podInfo.
+func (p *podInfo) parseStatus(status []string) error {
+	for _, line := range status {
+		tuples := strings.SplitN(line, "=", 2)
+		if len(tuples) != 2 {
+			glog.Warningf("Invalid status line: %q", line)
+			continue
+		}
+		switch tuples[0] {
+		case "state":
+			p.state = tuples[1]
+		case "networks":
+			p.networkInfo = tuples[1]
+		case "pid":
+			pid, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				glog.Errorf("Cannot parse pid %q: %v", tuples[1], err)
+				continue
+			}
+			p.pid = pid
+		}
+		if strings.HasPrefix(tuples[0], "sha512") {
+			if p.exitCodes == nil {
+				p.exitCodes = make(map[string]int)
+			}
+			exitcode, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				glog.Errorf("Cannot parse exit code %q: %v", tuples[1], err)
+			} else {
+				p.exitCodes[tuples[0]] = exitcode
+			}
+		}
+	}
+	return nil
+}
+
+// getIP returns the IP of a pod by parsing the network info.
+// The network info looks like this:
+//
+// default:ip4=172.16.28.3, database:ip4=172.16.28.42
+//
+func (p *podInfo) getIP() string {
+	parts := strings.Split(p.networkInfo, ",")
+
+	for _, part := range parts {
+		if strings.HasPrefix(part, "default:") {
+			return strings.Split(part, "=")[1]
+		}
+	}
+	return ""
+}
+
+// getContainerStatus creates the api.containerStatus of a container from the podInfo.
+// TODO(yifan): Get more detailed info such as Image, ImageID, etc.
+func (p *podInfo) getContainerStatus(container *kubecontainer.Container) api.ContainerStatus {
+	var status api.ContainerStatus
+	status.Name = container.Name
+	status.Image = container.Image
+
+	containerID, _ := parseContainerID(string(container.ID))
+	status.ImageID = containerID.imageID
+
+	switch p.state {
+	case Running:
+		// TODO(yifan): Get StartedAt.
+		status.State = api.ContainerState{
+			Running: &api.ContainerStateRunning{
+				StartedAt: util.Unix(container.Created, 0),
+			},
+		}
+	case Embryo, Preparing, Prepared:
+		status.State = api.ContainerState{Waiting: &api.ContainerStateWaiting{}}
+	case AbortedPrepare, Deleting, Exited, Garbage:
+		status.State = api.ContainerState{
+			Termination: &api.ContainerStateTerminated{
+				ExitCode:  p.exitCodes[status.ImageID],
+				StartedAt: util.Unix(container.Created, 0),
+			},
+		}
+	default:
+		glog.Warningf("Unknown pod state: %q", p.state)
+	}
+	return status
+}
+
+// toPodStatus converts a podInfo type into an api.PodStatus type.
+func (p *podInfo) toPodStatus(pod *kubecontainer.Pod) api.PodStatus {
+	var status api.PodStatus
+	status.PodIP = p.getIP()
+	// For now just make every container's state as same as the pod.
+	for _, container := range pod.Containers {
+		status.ContainerStatuses = append(status.ContainerStatuses, p.getContainerStatus(container))
```

Comment:
Containers are sharing fate at this moment, so they have the same status. This is something under hot discussion. Should be more clearer in the following weeks:
https://github.com/appc/spec/issues/276